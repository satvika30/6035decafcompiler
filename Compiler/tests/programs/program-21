class Program {
	int a[10];
	int num_permutations;

	void random_permutation() {
		// initialize
		int i;
		for i = 0, 10 {
			a[i] = i;
		}
		// permute randomly
		for i = 0, 10 {
			int k, j, temp;
			k = 9 - i;
			j = callout("random") % (k+1);
			temp = a[j];
			a[j] = a[k];
			a[k] = temp;
		}
	}

	// print values
	void print_array() {
		int i;
		for i = 0, 10 {
			callout("printf","%d ",a[i]);
		}
		callout("printf","\n");
	} 
	
	void main() {
		int i;
		callout("srandom", 17);
		num_permutations = 20;
		for i = 0, num_permutations {
			random_permutation();
			print_array();
		}
	}
}