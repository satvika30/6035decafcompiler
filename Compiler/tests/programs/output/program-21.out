6 3 0 9 8 1 2 7 4 5 
9 2 4 5 7 0 8 3 6 1 
4 8 7 0 3 1 9 5 2 6 
1 7 5 3 2 9 8 0 4 6 
3 9 4 7 0 5 6 2 8 1 
3 2 7 8 6 4 9 5 1 0 
6 9 3 5 0 8 4 1 7 2 
5 9 8 7 1 6 0 2 4 3 
0 4 2 5 3 1 6 9 8 7 
4 5 3 0 9 6 2 8 7 1 
4 8 9 6 2 5 1 7 0 3 
0 4 2 6 8 1 5 9 7 3 
6 2 3 7 0 9 8 1 5 4 
9 2 8 0 1 7 5 6 4 3 
6 7 0 1 8 9 3 4 5 2 
2 1 3 8 5 0 9 4 6 7 
2 9 0 1 8 6 7 3 4 5 
3 1 6 4 7 8 5 2 9 0 
4 3 7 8 1 6 9 2 5 0 
7 4 1 5 8 6 3 0 2 9